tests/unittests/core/worker/test_experiment.py

```diff
@@ -611,21 +609,160 @@ def test_configurable_broken_property(space: Space):
+        assert stats.trial_status_count == {"completed": NUM_COMPLETED, "reserved": 2}
+        # If max trials is infinite, then progress and ETA cannot be computed
+        assert stats.max_trials == "infinite"
+        assert stats.progress is None
+        assert stats.eta is None
+        assert stats.eta_milliseconds is None
+
+
+def test_experiment_stats_max_trials_zero(space: Space):
+    """Check stats with max trials 0."""
+    NUM_COMPLETED = 3
+    statuses = (["completed"] * NUM_COMPLETED) + (["reserved"] * 2)
+    with OrionState(trials=generate_trials(statuses)) as cfg:
+        exp = Experiment("supernaekei", mode="x", space=space, storage=cfg.storage)
+        exp._id = cfg.trials[0]["experiment"]
+        exp.metadata = {"datetime": datetime.datetime.utcnow()}
+        exp.max_trials = 0
+        stats = exp.stats
+        assert stats.trials_completed == NUM_COMPLETED
+        assert stats.best_trials_id == cfg.trials[2]["id"]
+        assert stats.best_evaluation == 0
+        assert stats.start_time == exp.metadata["datetime"]
+        assert stats.finish_time == cfg.trials[3]["end_time"]
+        assert stats.duration == datetime.timedelta(seconds=3)
+        assert stats.whole_clock_time == datetime.timedelta(seconds=3)
+        assert stats.nb_trials == NUM_COMPLETED + 2
+        assert stats.trial_status_count == {"completed": NUM_COMPLETED, "reserved": 2}
+        # If max trials is zero, then ETA cannot be computed, and progress is relative to nb trials
+        assert stats.max_trials == 0
+        assert stats.progress == 0.6
+        assert stats.eta is None
+        assert stats.eta_milliseconds is None
+
+
+def test_experiment_stats_max_trials_lesser_than_nb_completed(space: Space):
+    """Check stats with max trials < number of completed trials."""
+    NUM_COMPLETED = 3
+    statuses = (["completed"] * NUM_COMPLETED) + (["reserved"] * 2)
+    with OrionState(trials=generate_trials(statuses)) as cfg:
+        exp = Experiment("supernaekei", mode="x", space=space, storage=cfg.storage)
+        exp._id = cfg.trials[0]["experiment"]
+        exp.metadata = {"datetime": datetime.datetime.utcnow()}
+        exp.max_trials = 2
+        stats = exp.stats
+        assert stats.trials_completed == NUM_COMPLETED
+        assert stats.best_trials_id == cfg.trials[2]["id"]
+        assert stats.best_evaluation == 0
+        assert stats.start_time == exp.metadata["datetime"]
+        assert stats.finish_time == cfg.trials[3]["end_time"]
+        assert stats.duration == datetime.timedelta(seconds=3)
+        assert stats.whole_clock_time == datetime.timedelta(seconds=3)
+        assert stats.nb_trials == NUM_COMPLETED + 2
+        assert stats.trial_status_count == {"completed": NUM_COMPLETED, "reserved": 2}
+        # If max trials < completed trials, then ETA is 0, and progress is relative to nb trials
+        assert stats.max_trials == 2
+        assert stats.progress == 0.6
```

Comment:
This test should include new, broken and interrupted trials as well. The progress should not take into account the broken trials, because they will not be executed anymore, but the others yes.

Reply:
Ok !